components/brave_account/endpoint_client/response.h

```diff
@@ -0,0 +1,29 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ACCOUNT_ENDPOINT_CLIENT_RESPONSE_H_
+#define BRAVE_COMPONENTS_BRAVE_ACCOUNT_ENDPOINT_CLIENT_RESPONSE_H_
+
+#include <optional>
+
+#include "base/types/expected.h"
+#include "brave/components/brave_account/endpoint_client/is_response_body.h"
+#include "net/base/net_errors.h"
+
+namespace brave_account::endpoint_client {
+
+template <detail::IsResponseBody T, detail::IsResponseBody E>
+struct Response {
+  using SuccessBody = T;
+  using ErrorBody = E;
+
+  int error = net::ERR_IO_PENDING;
```

Comment:
nit: could be named `net_error` to reduce confusion, and match the name of its source

Reply:
https://github.com/brave/brave-core/pull/32336/commits/f735f927f348f19159398af7d9bf4a9762ae33b2, https://github.com/brave/brave-core/pull/32336/commits/df59090f71418ddc3d4bf106a1d6114bbfd4221e